recipe.cake

```diff
@@ -132,6 +167,111 @@ Task("Prepare-Chocolatey-Packages")
+            .WithProperty("OutputPath", MakeAbsolute(new DirectoryPath(BuildParameters.Paths.Directories.PublishedApplications + "/choco-no7zip/")).FullPath)
+            .WithProperty("TreatWarningsAsErrors", BuildParameters.TreatWarningsAsErrors.ToString())
+            .WithTarget("Build")
+            .SetMaxCpuCount(ToolSettings.MaxCpuCount)
+            .SetConfiguration("ReleaseOfficialNo7zip")
+            .WithLogger(
+                Context.Tools.Resolve("MSBuild.ExtensionPack.Loggers.dll").FullPath,
+                "XmlFileLogger",
+                string.Format(
+                    "logfile=\"{0}\";invalidCharReplacement=_;verbosity=Detailed;encoding=UTF-8",
+                    no7zLogPath
+                )
+            );
+
+        MSBuild(BuildParameters.SolutionFilePath, msbuildSettings);
+    }
+
+    if (FileExists(no7zLogPath))
+    {
+        BuildParameters.BuildProvider.UploadArtifact(no7zLogPath);
+    }
+}));
+
+Task("Prepare-ChocolateyNo7zip-Package")
+    .WithCriteria(() => BuildParameters.Configuration == "ReleaseOfficial", "Skipping No7zip because this isn't an official release")
+    .IsDependentOn("Build-ChocolateyNo7zip")
+    .IsDependeeOf("Sign-Assemblies")
+    .IsDependeeOf("Sign-PowerShellScripts")
+    .IsDependeeOf("Pack-ChocolateyNo7zip-Package")
+    .Does(() =>
+{
+    var nuspecDirectory = BuildParameters.Paths.Directories.ChocolateyNuspecDirectory + "-no7zip";
+    // Copy the Nuget/Chocolatey directory from Root Folder to temp/nuspec/chocolatey-no7zip
+    EnsureDirectoryExists(nuspecDirectory);
+    CopyFiles(GetFiles("./nuspec/chocolatey/**/*"), nuspecDirectory, true);
+    
+    // Copy legal documents
+    CopyFile(BuildParameters.RootDirectoryPath + "/docs/legal/CREDITS.md", nuspecDirectory + "/tools/chocolateyInstall/CREDITS.txt");
+
+    // Run Chocolatey Unpackself
+    CopyFile(BuildParameters.Paths.Directories.PublishedApplications + "/choco-no7zip_merged/choco.exe", nuspecDirectory + "/tools/chocolateyInstall/choco.exe");
+
+    StartProcess(nuspecDirectory + "/tools/chocolateyInstall/choco.exe", new ProcessSettings{ Arguments = "unpackself -f -y --allow-unofficial-build" });
+
+    // Tidy up logs and config folder which are not required
+    var logsDirectory = nuspecDirectory + "/tools/chocolateyInstall/logs";
+    var configDirectory = nuspecDirectory + "/tools/chocolateyInstall/config";
+
+    if (DirectoryExists(logsDirectory))
+    {
+        DeleteDirectory(logsDirectory, new DeleteDirectorySettings {
+            Recursive = true,
+            Force = true
+        });
+    }
+
+    if (DirectoryExists(configDirectory))
+    {
+        DeleteDirectory(configDirectory, new DeleteDirectorySettings {
+            Recursive = true,
+            Force = true
+        });
+    }
+});
+
+Task("Pack-ChocolateyNo7zip-Package")
+    .WithCriteria(() => BuildParameters.Configuration == "ReleaseOfficial", "Skipping No7zip because this isn't an official release")
```

Comment:
We should likely have some additional criteria here:

```
.WithCriteria(() => BuildParameters.ShouldRunChocolatey, "Skipping because execution of Chocolatey has been disabled")
.WithCriteria(() => BuildParameters.BuildAgentOperatingSystem == PlatformFamily.Windows, "Skipping because not running on Windows")
```